### DISCUSSION TITLE
How to optimize images in Wordpress content ?

### DISCUSSION THREAD

**Discussion by chawax** [vercel/next.js#40478] [Help] [answered]
Hi all,

I use Next.js with headless Wordpress but I have problems to optimize images in a Wordpress content. Wordpress generates a `<figure>` node in HTML, so I wrote a regex to extract those figures and replace them with Next.js `Image` component result. Something like this : 

```
function replaceImgWithNextImage(content: string, defaultAlt?: string): string {
  // We use a regex to extract width, src and alt
  return content.replaceAll(
    /<figure .*width="([^"]*)".*src="([^"].*)".*alt="([^"]*)".*<\/figure>/gm,
    (_args, _width, src, alt) => {
      // Image source url contains a size set by Wordpress. For example here : 1024x681
      // We have to remove it to get the original image name in Wordpress

      const output = ReactDOMServer.renderToStaticMarkup(
        <React.Fragment>
          <Image
            src={src}
            alt={alt}
            title={alt}
            width={1200}
            height={800}
            loader={wordpressImageLoader}
          />
        </React.Fragment>
      );

      return output;
    }
  );
}
```

But the HTML code this generates looks wrong, with a Base64 image in `img` tag and what I expect to be in `img` tag is with `noscript` tags. Something like this (I simplified the outptut, removing style attribute to make it more readable) :

```
<span>
     <span><img alt="" aria-hidden="true" src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271200%27%20height=%27800%27/%3e"></span>
     <img alt="xxx" title="xxx" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" decoding="async" data-nimg="intrinsic">
     <noscript>
          <img alt="xxx" 
                title="xxx"
                srcSet="https://assets.xxx.com/cms/image-1200.jpg 1x, https://assets.xxx.com/cms/image-3840.jpg 2x" src="https://assets.xxx.com/cms/image-3840.jpg"
                decoding="async"
                data-nimg="intrinsic"
                loading="lazy"/>
      </noscript>
</span>
```

Any idea what I am doing wrong ? Is `Image` component safe to be used with `ReactDOMServer.renderToStaticMarkup` ? Is there a easier and better way to do this kind of thing ?

**Comment by styfle** [comment, marked as answer]
Yes that is the expected HTML for `next/image`.

Try changing `next/image` to `next/future/image` if you want to remove the wrapper.

This will become the default behavior in Next.js 13 but you can use it today with Next.js 12.3 👍 

https://nextjs.org/blog/next-12-3

**Reply by chawax** [reply]
That's exactly what I needed ! Thanks a lot !
